app/controllers/SummaryController.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import config.ErrorHandler
+import controllers.actions.{AuthenticatedAction, JourneyRefiner}
+import javax.inject.Inject
+import models.cache.{ArrivalAnswers, DepartureAnswers, JourneyType}
+import play.api.Logger
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.MovementRepository
+import services.SubmissionService
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import views.html.movement_confirmation_page
+import views.html.summary.{arrival_summary_page, departure_summary_page}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class SummaryController @Inject()(
+  authenticate: AuthenticatedAction,
+  getJourney: JourneyRefiner,
+  errorHandler: ErrorHandler,
+  movementRepository: MovementRepository,
+  submissionService: SubmissionService,
+  mcc: MessagesControllerComponents,
+  arrivalSummaryPage: arrival_summary_page,
+  departureSummaryPage: departure_summary_page,
+  movementConfirmationPage: movement_confirmation_page
+)(implicit ec: ExecutionContext)
+    extends FrontendController(mcc) with I18nSupport {
+
+  private val logger = Logger(this.getClass)
+
+  def displayPage(): Action[AnyContent] = (authenticate andThen getJourney(JourneyType.ARRIVE, JourneyType.DEPART)) { implicit request =>
+    request.answers match {
+      case arrivalAnswers: ArrivalAnswers     => Ok(arrivalSummaryPage(arrivalAnswers))
+      case departureAnswers: DepartureAnswers => Ok(departureSummaryPage(departureAnswers))
+    }
+  }
+
+  def submitMovementRequest(): Action[AnyContent] = (authenticate andThen getJourney(JourneyType.ARRIVE, JourneyType.DEPART)).async {
+    implicit request =>
+      submissionService
+        .submitMovementRequest(request.pid, request.answers)
+        .flatMap {
+          case (Some(consignmentReferences), ACCEPTED) =>
+            movementRepository.delete(request.pid).map { _ =>
+              Ok(movementConfirmationPage(consignmentReferences))
+            }
+          case _ =>
+            Future.successful {
```

Comment:
We shouldn't be handling this kind of thing in Controllers, errors should bubble up to the error handler

I'd suggest we alter the SubmissionService method to just return `Future[Option[ConsignmentReferences]]`

Reply:
Just copied from movements

Reply:
Changed service to return Future[ConsigmentReference] and no longer handle the error